File: process/Status.ts
/// Determines the current status of the process
/// (ex: if it is running, waiting, etc)
enum Status {
  /// Process is currently being run in a processor
  Running,
  /// Process is ready to be run in a processor, but is not currently being run
  Ready,
  /// Process cannot be run in the processor for some reason
  Blocked,
  /// Process is freshly created (this should be the default)
  New
}

export default Status

File: proc/Status.go
package proc

// Status describes the status of a given process
type Status int

const (
	// New describes a fresh, newly-created process
	New Status = iota
	// Ready describes a process that is ready to do something
	Ready
	// Blocked describes a process that is currently held up waiting on something
	Blocked
	// Running describes a process in the middle of execution
	Running
)
